src/signals/incident/components/form/FileInput/index.js

```diff
@@ -2,56 +2,123 @@ import React from 'react';
 import PropTypes from 'prop-types';
 
 import Header from '../Header/';
-import { validateFileType, validateMaxFilesize } from '../../../services/custom-validators';
+import fileSize from '../../../services/file-size';
+
 import './style.scss';
 
+const ERROR_TIMEOUT_INTERVAL = 8000;
+
 const FileInput = ({ handler, touched, hasError, getError, parent, meta, validatorsOrOpts }) => {
+  let timeoutInstance = null;
+  const maxNumberOfFiles = (meta && meta.maxNumberOfFiles) || 3;
   const handleChange = (e) => {
+    /* istanbul ignore next */
     if (e.target.files && e.target.files.length) {
-      const file = e.target.files[0];
+      const maxFileSizeFilter = meta.maxFileSize ? checkFileSize : () => true;
+      const allowedFileTypesFilter = meta.allowedFileTypes ? checkFileType : () => true;
+      const maxNumberOfFilesFilter = checkNumberOfFiles;
+      const existingFiles = handler().value || [];
+      const existingPreviews = (parent && parent.value && parent.value[`${meta.name}_previews`]) || [];
+      const batchFiles = [...e.target.files];
+
+      existingFiles.map((file) => {
+        // eslint-disable-next-line no-param-reassign
+        file.existing = true;
+        return file;
```

Comment:
You could do `return { ...file, existing: true }` to prevent having to disable the linter

Reply:
gedaan